controller/v1.10.1/rockcraft.yaml

```diff
@@ -0,0 +1,801 @@
+# registry.k8s.io/ingress-nginx/controller:v1.10.1
+
+name: controller
+summary: Rock for the Nginx ingress controller.
+description: |
+  Rock for the Nginx ingress controller.
+  https://github.com/kubernetes/ingress-nginx
+license: Apache-2.0
+
+version: v1.10.1
+
+# Upstream image is based on Alpine:
+# https://github.com/kubernetes/ingress-nginx/blob/controller-v1.10.1/images/nginx-1.25/rootfs/Dockerfile#L24
+base: ubuntu@22.04
+build-base: ubuntu@22.04
+platforms:
+  amd64:
+  arm64:
+
+services:
+  nginx:
+    startup: enabled
+    override: replace
+
+    command: nginx -g 'daemon off;'
+
+# https://github.com/kubernetes/ingress-nginx/blob/controller-v1.10.1/images/nginx-1.25/rootfs/Dockerfile#L26-L29
+environment:
+  # NOTE: string interpolation is not supported within 'environment' variable delcarations.
+  # This value is simply the default $PATH on Ubuntu 22.04 images with the
+  # relevant lua/nginx binary locations appended to it:
+  # https://github.com/kubernetes/ingress-nginx/blob/controller-v1.10.1/images/nginx-1.25/rootfs/Dockerfile#L26
+  PATH: /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/local/luajit/bin:/usr/local/nginx/sbin:/usr/local/nginx/bin
+
+  LUA_PATH: /usr/local/share/luajit-2.1.0-beta3/?.lua;/usr/local/share/lua/5.1/?.lua;/usr/local/lib/lua/?.lua;;
+  LUA_CPATH: /usr/local/lib/lua/?/?.so;/usr/local/lib/lua/?.so;;
+  LD_LIBRARY_PATH: /usr/local/lib
+
+parts:
+
+  prepare-base-image:
+    plugin: nil
+    stage-packages:
+      # https://github.com/kubernetes/ingress-nginx/blob/controller-v1.10.1/images/nginx-1.25/rootfs/Dockerfile#L39-L53
+      - bash
+      - passwd
+      - libpcre3
+      - libpcre3-dev
+      - zlib1g-dev
+      - ca-certificates
+      - patch
+      - libyajl2
+      - yajl-tools
+      - liblmdb-dev
+      - lmdb-utils
+      - libxml2
+      - libmaxminddb-dev
+      - libyaml-cpp-dev
+      - libprotobuf-dev
+      - dumb-init
+      - tzdata
+
+    overlay-script: |
+      set -eux -o pipefail
+
+      # https://github.com/kubernetes/ingress-nginx/blob/controller-v1.10.1/images/nginx-1.25/rootfs/Dockerfile#L55-L56
+      # NOTE(aznashwan): the base Ubuntu image already defines a 'www-data' user.
+      # groupadd -R $CRAFT_OVERLAY -r -g 101 www-data
+      # useradd -R $CRAFT_OVERLAY \
+      #   -s /sbin/nologin -d /usr/local/nginx --no-log-init -r -m -g 101 -u 101 www-data
+
+  build-libgrpcpp:
+    plugin: nil
+    after: ['prepare-base-image']
+    build-packages:
+      - cmake
+    override-build: |
+      set -eux -o pipefail
+
+      # HACK(aznashwan): the `libgrpc++` package on 22.04 is version 1.30,
+      # while the `grpc-dev` package in the Alpine repos used in the original
+      # image is currently 1.62.1, so we must build it ourselves:
+      LIBGRPC_VERSION=v1.62.1
+
+      cd /tmp
+      git clone --recurse-submodules --depth 1 --shallow-submodules \
+        -b $LIBGRPC_VERSION https://github.com/grpc/grpc
```

Comment:
Wouldn't it be better if we defined the source in the part with `source: https://github.com/grpc/grpc`?